### PR TITLE
archie/fix_paths

### PR DESCRIPTION
**Pull request by archiepeterjarvis**
# Use relative paths and Kivy resources to speed up startup 

[Jira Ticket](https://yetitoolsmartsoftware.atlassian.net/jira/software/projects/YTSS/boards/1?assignee=5f8db15e6bc6340068c0bfc3&selectedIssue=YTSS-2862)
- [x] Ready for review

## Checklist
- [x] I have completed a self review
- [x] I have set the recent milestone
- [x] I have updated the jira ticket
- [x] I have added the relevant labels to this PR
- [ ] I have updated documentation (if applicable)
- [ ] I have run the unit tests suite and they pass
- [ ] I have updated the [change log](https://docs.google.com/document/d/1t3s6lQuUcug1Fj8E_i2gBJNvXN0-funwBArG2hsia6k/edit)

## Description
After profiling easycut, I found the path_utils module taking 4+ seconds on startup. This should reduce it to less than a second I would expect. 

## Notes
Not implemented into existing usages of path_utils, just setup for usage.

## Testing
If you run paths.py, it will output the path to a resource and print the paths out to test. Ensure your working directory is set correctly!!

### Visual Test
- [ ] Not applicable
- [ ] Own computer
- [ ] Console 7"
- [ ] Console 10"

### Function Test
- [ ] Not applicable
- [x] Own computer
- [ ] Console 7"
- [ ] Console 10"

### Unit Tests
- [ ] Not applicable
- [ ] Completed

## Screenshots (if applicable)

### CODE REVIEW

File: src/asmcnc/paths.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+def create_paths():
+    if not os.path.exists(SB_VALUES_PATH):
+        Logger.info("SB values directory not found, creating...")
+        os.makedirs(SB_VALUES_PATH)
+
+
+# Register any paths that contain resources
+resource_add_path(SKAVA_UI_IMG_PATH)
+resource_add_path(SHAPE_CUTTER_IMG_PATH)
```
Comment (Masc999) [inline src/asmcnc/paths.py:48]: Other img paths are missing.  There are 15 when I search for "img" within the src folder.

File: src/asmcnc/paths.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+SKAVA_UI_IMG_PATH = os.path.join(SKAVA_UI_PATH, "img")
+SHAPE_CUTTER_IMG_PATH = os.path.join(ASMCNC_DIR, "apps", "shapeCutter_app", "img")
+
+# Paths that may need to be created
+SB_VALUES_PATH = os.path.join(ROOT_DIR, "sb_values")
+
+
+def create_paths():
+    if not os.path.exists(SB_VALUES_PATH):
+        Logger.info("SB values directory not found, creating...")
```
Comment (Masc999) [inline src/asmcnc/paths.py:42]: I would make this a Logger.warning

File: src/asmcnc/paths.py (hunk 1 above)
Comment (archiepeterjarvis) [reply src/asmcnc/paths.py:48]: This is POC not entirely implemented to ensure you were happy :)

File: src/asmcnc/apps/drywall_cutter_app/engine.py [hunk 3]
```diff
@@ -565,9 +565,9 @@ def get_custom_shape_extents(self):
 
     # Main
     def engine_run(self, simulate=False):
-        temp_gcode_path = pu.get_path('drywall_cutter_app/config/temp', folders_only=True)
+        temp_gcode_path = os.path.join(self.source_folder_path, "temp")
```
Comment (Masc999) [inline src/asmcnc/apps/drywall_cutter_app/engine.py:568]: This folder does not exist and needs to be created!

File: src/asmcnc/core_UI/components/buttons/vacuum_button.py [hunk 4]
```diff
@@ -5,14 +5,13 @@
...
-from asmcnc.core_UI import path_utils
+from asmcnc import paths
 from asmcnc.core_UI.components.buttons.button_base import ImageButtonBase
 from asmcnc.core_UI.components.widgets.blinking_widget import FastBlinkingWidget
 
-SKAVA_UI_PATH = path_utils.get_path("skavaUI")
-SKAVA_UI_IMG_PATH = os.path.join(SKAVA_UI_PATH, "img")
-EXTRACTOR_IMAGE = os.path.join(SKAVA_UI_IMG_PATH, "extraction_on.png")
-RED_NO_SIGN_IMAGE = os.path.join(SKAVA_UI_IMG_PATH, "off_icon.png")
+SKAVA_UI_IMG_PATH = os.path.join(paths.SKAVA_UI_IMG_PATH, "img")
```
Comment (Masc999) [inline src/asmcnc/core_UI/components/buttons/vacuum_button.py:12]: This line can be deleted.

File: src/asmcnc/paths.py [hunk 5]
```diff
@@ -2,12 +2,43 @@
...
+    DATA_CONSENT_APP_PATH: Path to the data_consent_app directory. (easycut-smartbench/src/asmcnc/apps/start_up_sequence/data_consent_app)
+    DATA_CONSENT_IMG_PATH: Path to the img directory in data_consent_app. (easycut-smartbench/src/asmcnc/apps/start_up_sequence/data_consent_app/img)
+    WARRANTY_APP_PATH: Path to the warranty_app directory. (easycut-smartbench/src/asmcnc/apps/start_up_sequence/warranty_app)
+    WARRANTY_IMG_PATH: Path to the img directory in warranty_app. (easycut-smartbench/src/asmcnc/apps/start_up_sequence/warranty_app/img)
+    WELCOME_TO_SMARTBENCH_APP_PATH: Path to the welcome_to_smartbench_app directory. (easycut-smartbench/src/asmcnc/apps/start_up_sequence/welcome_to_smartbench_app)
+    WELCOME_TO_SMARTBENCH_IMG_PATH: Path to the img directory in welcome_to_smartbench_app. (easycut-smartbench/src/asmcnc/apps/start_up_sequence/welcome_to_smartbench_app/img)
+    JOB_GO_IMG_PATH: Path to the img directory in job_go. (easycut-smartbench/src/asmcnc/core_UI/job_go/img)
+    SEQUENCE_ALARM_IMG: Path to the img directory in sequence_alarm. (easycut-smartbench/src/asmcnc/core_UI/sequence_alarm/img)
```
Comment (Masc999) [inline src/asmcnc/paths.py:38]: Naming: Should be ...IMG_PATH

File: src/asmcnc/paths.py [hunk 6]
```diff
@@ -25,16 +57,62 @@
...
+SYSTEM_TOOLS_IMG_PATH = os.path.join(SYSTEM_TOOLS_APP_PATH, "img")
+
+UPGRADE_APP_PATH = os.path.join(APPS_PATH, "upgrade_app")
+UPGRADE_IMG_PATH = os.path.join(UPGRADE_APP_PATH, "img")
+
+WIFI_APP_PATH = os.path.join(APPS_PATH, "wifi_app")
+WIFI_IMG_PATH = os.path.join(WIFI_APP_PATH, "img")
+
+START_UP_SEQUENCE_PATH = os.path.join(APPS_PATH, "start_up_sequence")
+START_UP_SEQUENCE_IMG = os.path.join(START_UP_SEQUENCE_PATH, "screens", "img")
```
Comment (Masc999) [inline src/asmcnc/paths.py:99]: same here. Should be ...IMG_PATH

**Review by Masc999 — APPROVED** [review]

**Pull request merged**